collagraph/components/__init__.py

```diff
@@ -3,23 +3,28 @@
 from observ import reactive, readonly
 
 
-RENDER_FUNCTION = None
-
-
 def set_render_function(fun):
     global RENDER_FUNCTION
     RENDER_FUNCTION = fun
 
 
-class Meta(type):
+class ComponentMeta(type):
+    """
+    Meta class for inserting a render function into a class
+    from a single-file component, whenever the RENDER_FUNCTION
+    class attribute has been set.
+    """
+
+    RENDER_FUNCTION = None
+
     def __new__(cls, name, bases, attrs):
-        if RENDER_FUNCTION:
-            attrs["render"] = RENDER_FUNCTION
+        if cls.RENDER_FUNCTION:
+            attrs["render"] = cls.RENDER_FUNCTION
```

Comment:
You could raise an error if render is already in attrs as a safety rail for users